pkg/odo/cli/utils/utils.go

```diff
@@ -36,3 +42,63 @@ func VisitCommands(cmd *cobra.Command, f func(*cobra.Command)) {
 		VisitCommands(child, f)
 	}
 }
+
+// LocalConfigInfo is a struct that will contain the LocalConfig as well as SourcePath
+// the reasoning behind SourcePath is due to the fact that we must correct what the path is
+// on different platforms (Linux, Windows, etc.) as well as get the absolute path of the component.
+type LocalConfigInfo struct {
+	LocalConfig *config.LocalConfigInfo
+	SourcePath  string
+}
+
+// RetrieveLocalConfigInfo retrieves the local configuration
+func RetrieveLocalConfigInfo(componentContext string) (configInfo LocalConfigInfo, err error) {
```

Comment:
Can we move this into pkg/odo/genericclioptions/context.go ?
Essentially make localconfig a member field of Context and resolve it in pkg/odo/genericclioptions/context.go#newContext probably after removing project, app, component names from Context struct and resolving based on context flag in cobra.Command and hence the resolution will be there for every command ?